doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/defaultProcessors/IncludeDocProcessor.kt

```diff
@@ -0,0 +1,263 @@
+        ) { it != documentable }
+
+        if (queried == null) {
+            val queriedNoFilter = documentable.queryDocumentables(
+                query = includePath,
+                documentables = filteredDocumentables,
+            )
+            val attemptedQueries = documentable.getAllFullPathsFromHereForTargetPath(includePath)
+                .joinToString("\n")
+
+            val queriedPath = documentable.queryDocumentablesForPath(
+                query = includePath,
+                documentables = allDocumentables,
+            )
+
+            error(
+                when {
+                    queriedNoFilter == documentable ->
+                        "IncludeDocProcessor ERROR: Self-reference detected. Called from \"$path\" in \"${documentable.file.absolutePath}\"."
+
+                    queriedPath != null ->
+                        "IncludeDocProcessor ERROR: Include path found, but no documentation found for: " +
+                                "\"$includePath\". Including documentation from outside the library or from type-aliases " +
+                                "is currently not supported.\nCalled from \"$path\" in \"${documentable.file.absolutePath}\".\n" +
+                                "Attempted queries: [\n$attemptedQueries]\n" +
+                                "Include line: \"$line\"\n" +
+                                "Full doc: \"${documentable.docContent}\""
+
+                    else ->
+                        "IncludeDocProcessor ERROR: Include not found: \"$includePath\".\nCalled from \"$path\" in \"${documentable.file.absolutePath}\".\n" +
+                                "Attempted queries: [\n$attemptedQueries]\n"+
+                                "Include line: \"$line\"\n" +
+                                "Full doc: \"${documentable.docContent}\""
+                }
+            )
+        }
+
+        var content: DocContent = queried.docContent
+            .removePrefix("\n")
+            .removeSuffix("\n")
+
+        if (extraContent.isNotBlank()) {
+            content = "$content $extraContent"
+        }
+
+        content = when {
+
+            // if the content contains links to other elements, we need to expand the path
+            // providing the original name or alias as new alias.
+            documentable.isKotlin -> content.replaceKdocLinks { query ->
+                queried.queryDocumentablesForPath(
+                    query = query,
+                    documentables = allDocumentables,
+                    pathIsValid = { path, it ->
+
+                        // ensure that the given path points to the same element in the destination place and
+                        // that it's queryable
+                        documentable.queryDocumentables(
+                            query = path,
+                            documentables = allDocumentables,
+                        ) == it
+                    },
+                ) ?: query
+            }
+
```

Comment:
isKotlin ->
isJava ->
?

Reply:
Does it really works well on the Java? If not, probably it will be better to remove Java support from the first version

Reply:
I´ll make it an enum, it's mutually exclusive anyways.

Yes, Java is definitely supported. I take extra care of escaping unsupported characters, using `@inlcude {@link KotlinFileKt#someFun}` from Java, etc. However, due to the complexity of Java link statements compared to Kotlin `[references]`, I don't yet support expanding those to their fully qualified path whenever a statement like that is copied over to other docs. Hence warning.